core/imagedownloader.cpp

```diff
@@ -39,8 +39,11 @@ void ImageDownloader::load(QString filename, bool fromHash)
 {
 	QUrl url = fromHash ? cloudImageURL(qPrintable(filename)) : QUrl::fromUserInput(filename);
 
-	if (!url.isValid())
+	// If this is a file, we tried previously -> don't bother trying it again
+	if (url.scheme() == "file" || !url.isValid()) {
```

Comment:
It's fine like this, but you could have also written url.isLocalFile()

Reply:
Damn - missed that function yesterday night. :( That is indeed much nicer. Care to submit a cleanup-PR?